iceoryx_posh/source/popo/wait_set.cpp

```diff
@@ -107,23 +103,33 @@ WaitSet::ConditionVector WaitSet::createVectorWithFullfilledConditions() noexcep
     {
         if (currentCondition->hasTriggered())
         {
-            if (!conditions.push_back(currentCondition))
-            {
-                errorHandler(Error::kPOPO__WAITSET_CONDITION_VECTOR_OVERFLOW, nullptr, ErrorLevel::FATAL);
-            }
+            // We do not need to verify if push_back was successful since
+            // m_conditionVector and conditions are having the same type, a
+            // vector with the same guaranteed capacity.
+            // Therefore it is guaranteed that push_back works!
+            static_cast<void>(conditions.push_back(currentCondition));
         }
     }
+
     return conditions;
 }
 
 template <typename WaitFunction>
 WaitSet::ConditionVector WaitSet::waitAndReturnFulfilledConditions(const WaitFunction& wait) noexcept
 {
-    /// @note Inbetween here and last wait someone could have set the trigger to true, hence reset it
-    m_conditionVariableWaiter.reset();
+    WaitSet::ConditionVector conditions;
+
+    if (m_conditionVariableWaiter.wasTriggered())
```

Comment:
Just as info: The WaitSet itself is not threadsafe. 

But it is possible that after `wasTriggered` returned false a condition is triggered. In this case the method would return an empty condition vector and wait would instantly enter this method again and would create the vector with the triggered condition.

The other case that `wasTriggered` returned true and after `reset()` a condition is triggered which is then contained in the returning `conditions` vector.
In this case we would enter the `wasTriggered` condition again in the next run but the would get an empty conditions vector. This would lead to a `wait` call where we wait for additional conditions. If there are none we would return an empty conditions vector.

I see no problem which those cases. Do you see any problems @MatthiasKillat

Reply:
I came to the conclusion that it behaves like before. The contract must be: 

- if there is no timeout then we return only with at least one triggered condition
- we do not wait if a condition was already triggered